### PR TITLE
Fixes compatibility issues with aws-php-sdk v3.

### PR DESCRIPTION
**Pull request by puckbag**
- Update README about `create-iam-user`. More details about AWS environment credentials and profiles would be nice.
- `Aws\Common\Aws` no longer exists.
- Replace deprecated `::factory` calls with proper instantiations.

### CODE REVIEW

File: inc/class-s3-uploads.php [hunk 1]
```diff
@@ -7,6 +7,7 @@ class S3_Uploads {
 	private        $bucket_url;
 	private        $key;
 	private        $secret;
+    private        $s3;
```
Comment (rmccue) [inline inc/class-s3-uploads.php]: Indentation here is a bit off; make sure you're using tabs instead of spaces.

**Comment by BandonRandon** [comment]
Just commenting to say that on my server running Debian Jesse 8 and PHP 7.1 I had to apply this PR/Patch to fix the following fatal:
```PHP Fatal error:  Uncaught TypeError: Argument 1 passed to S3_Uploads_Stream_Wrapper::register() must be an instance of Aws\S3\S3ClientInterface, instance of Aws\S3\S3Client given, called in plugins/s3-uploads/inc/class-s3-uploads.php on line 75 and defined in plugins/s3-uploads/inc/class-s3-uploads-stream-wrapper.php:105
```

This patch cleared things right up for me.
I also wanted to mention I had the same problem noted in  https://github.com/humanmade/S3-Uploads/issues/95#issuecomment-254318557 I was able to fix the conflict by copying the AWS SDK from S3-Uploads to BackupWordPress S3. I reported it https://github.com/humanmade/backupwordpress-pro-s3/issues/34 but since they are both HM products wanted to note here as well.

**Comment by epoplive** [comment]
I had to combine the old and new format command line's to get the command to run, but then I get an error <Message>Credential should be scoped to a valid region, not 'us-west-2'. </Message>

**Comment by hxpro** [comment]
Hi @joehoyle 
Please, can you also check `S3_Uploads_ChangedFilesIterator` ? It extends `Aws\S3\Sync\ChangedFilesIterator`. The `Aws\S3\Sync` namespace have been replaced by the `Aws\S3\Transfer` class in aws-php-sdk v3.

**Comment by joehoyle** [comment]
This is now fixed, as we dswitched ot loading the cdk via composer and fix the compat issues mentioned in this PR. Thanks all!

**Pull request closed without merging**